google/cloud/forseti/services/scanner/dao.py

```diff
@@ -18,150 +18,259 @@
-            """Create a session to read from the models table.
-
-            Returns:
-                ScopedSessionmaker: A scoped session maker that will create
-                    a session that is automatically released.
-            """
-            return db.ScopedSessionMaker(
-                sessionmaker(
-                    bind=self.engine,
-                    expire_on_commit=False),
-                auto_commit=True)
-
-        def create(self, violations, inventory_index_id):
-            """Save violations to the db table.
-
-            Args:
-                violations (list): A list of violations.
-                inventory_index_id (str): Id of the inventory index.
-            """
-            with self.violationmaker() as session:
-                created_at_datetime = date_time.get_utc_now_datetime()
-                for violation in violations:
-                    violation_hash = _create_violation_hash(
-                        violation.get('full_name', ''),
-                        violation.get('resource_data', ''),
-                        violation.get('violation_data', ''),
-                    )
-
-                    violation = self.TBL_VIOLATIONS(
-                        inventory_index_id=inventory_index_id,
-                        resource_id=violation.get('resource_id'),
-                        resource_type=violation.get('resource_type'),
-                        full_name=violation.get('full_name'),
-                        rule_name=violation.get('rule_name'),
-                        rule_index=violation.get('rule_index'),
-                        violation_type=violation.get('violation_type'),
-                        violation_data=json.dumps(
-                            violation.get('violation_data')),
-                        resource_data=violation.get('resource_data'),
-                        violation_hash=violation_hash,
-                        created_at_datetime=created_at_datetime
-                    )
-
-                    session.add(violation)
-
-        def list(self, inventory_index_id=None):
-            """List all violations from the db table.
-
-            Args:
-                inventory_index_id (str): Id of the inventory index.
-
-            Returns:
-                list: List of Violation row entry objects.
-            """
-            with self.violationmaker() as session:
-                if inventory_index_id:
-                    return (
-                        session.query(self.TBL_VIOLATIONS).filter(
-                            self.TBL_VIOLATIONS.inventory_index_id ==
-                            inventory_index_id).all()
-                    )
-                return (
-                    session.query(self.TBL_VIOLATIONS).all())
-
-    base.metadata.create_all(dbengine)
-
-    return ViolationAccess
+    scanner_index = None
+    if not index_state:
+        scanner_index = (
+            session.query(ScannerIndex)
+            .filter(and_(or_(
```

Comment:
Instead of `or_()`, you can also use the `in_()`, which would make the filter less repetitive and easier to extend in the future, something like below.
```
retrievable_states = [`SUCCESS`, `PARTIAL_SUCCESS`]
ScannerIndex.scanner_status.in_(retrievable_states)
```

Reply:
Done.